From 47795931f8bf6a2a380622a88c4e8506a858b1f7 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Wed, 7 Apr 2010 20:09:34 +0800
Subject: [PATCH 06/17] fix capabilities definition

For "capabilties^{}" lines, replace obj-id with zero-id.

Define capabilities using rules from pack-protocol.txt - ideally, this
should be shifted to protocol-common.txt.
---
 Documentation/technical/http-protocol.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/Documentation/technical/http-protocol.txt b/Documentation/technical/http-protocol.txt
index 483162f..22a22b4 100644
--- a/Documentation/technical/http-protocol.txt
+++ b/Documentation/technical/http-protocol.txt
@@ -257,14 +257,16 @@ declarations behind a NUL on the first ref.
 		      "0000"
   ref_list         =  empty_list / non_empty_list
 
-  empty_list       =  PKT-LINE(obj-id SP "capabilities^{}" NUL cap_list LF)
+  empty_list       =  PKT-LINE(zero-id SP "capabilities^{}" NUL cap-list LF)
 
   non_empty_list   =  PKT-LINE(obj-id SP name NUL cap_list LF)
 		      *ref_record
 
-  cap_list         =  *(SP capability) SP
-  ref_record       =  any_ref / peeled_ref
+  cap-list         =  capability *(SP capability)
+  capability       =  1*(LC_ALPHA / DIGIT / "-" / "_")
+  LC_ALPHA         =  %x61-7A
 
+  ref_record       =  any_ref / peeled_ref
   any_ref          =  PKT-LINE(obj-id SP name LF)
   peeled_ref       =  PKT-LINE(obj-id SP name LF)
 		      PKT-LINE(obj-id SP name "^{}" LF
-- 
1.8.4.rc4.527.g303b16c

